pybamm/expression_tree/symbol.py

```diff
@@ -901,7 +901,6 @@ def evaluates_to_number(self):
     def evaluates_to_constant_number(self):
         return self.evaluates_to_number() and self.is_constant()
 
-    @lru_cache
```

Comment:
Is this also removed while linting?

Reply:
Yes, to resolve the violation of the [rule cached-instance-method (B019)](https://docs.astral.sh/ruff/rules/cached-instance-method/). 

The `lru_cache` here is redundant as the function already handles the storage of previous results in `self._saved_evaluates_on_edges[dimension]`, which are returned in $O(1)$ time.